src/odemis/acq/move.py

```diff
@@ -234,12 +239,40 @@ def getCurrentAlignerPositionLabel(current_pos, align):
     return UNKNOWN
 
 
-def _getDistance(start, end):
-    # Calculate the euclidean distance between two 3D points
-    axes = start.keys() & end.keys()  # only the axes found on both points
-    sp = numpy.array([start[a] for a in axes])
-    ep = numpy.array([end[a] for a in axes])
-    return scipy.spatial.distance.euclidean(ep, sp)
+def _getDifference(start, end):
+    """
+    Calculate the error/difference between two 3D points.
+    start, end (dict -> float): a 3D point
+    return (float): the difference between two 3D points
+    """
+    axes = start.keys() & end.keys()
+    lin_axes = axes & {'x', 'y', 'z'}  # only the axes found on both points
+    rot_axes = axes & {'rx', 'rz'}  # only the axes found on both points
+    if not lin_axes and not rot_axes:
+        return
+    # for the linear error
+    sp = numpy.array([start[a] for a in lin_axes])
+    ep = numpy.array([end[a] for a in lin_axes])
+    lin_error = scipy.spatial.distance.euclidean(ep, sp)
+    # for the rotation error
+    Rx_start = Rx_end = Rz_start = Rz_end = numpy.eye(3)
+    for a in rot_axes:
+        if a == "rx":
+            # find the elemental rotation around x axis
+            Rx_start = Rotation.from_euler('x', start["rx"], degrees=False).as_matrix()
+            Rx_end = Rotation.from_euler("x", end['rx']).as_matrix()
+        elif a == "rz":
+            # find the elemental rotation around z axis
+            Rz_end = Rotation.from_euler("z", end['rz']).as_matrix()
+            Rz_start = Rotation.from_euler('z', start["rz"], degrees=False).as_matrix()
+    # find the total rotations
+    R_start = numpy.matmul(Rz_start, Rx_start)
+    R_end = numpy.matmul(Rz_end, Rx_end)
+    # find the rotation error matrix
+    R_diff = numpy.matmul(numpy.transpose(R_start), R_end)
+    # map to scalar error
+    rot_error = 0.03*numpy.trace(numpy.eye(3)-R_diff)
+    return lin_error+rot_error
```

Comment:
My eye came on this function again when i was reviewing the code for the last time before committing and pushing. The Euclidean distance can be calculated only between Cartesian points. It can not be calculated between angles. This explains the weird behavior of the gauge (the gauge does not progress if the stage moves along x, y or z, but progresses if the stage moves around rx or rz). Therefore the linear error and the rotation error must be calculated separately and then summed. The rotation error can be calculated as follows:

1. Find the elemental rotations of the stage.
2. Find the total rotation of the stage which is the multiplication of the elemental rotations
![R](https://user-images.githubusercontent.com/79858941/134505865-2a711b17-96ac-4d42-a6e9-082a1f080246.png)
Luckily, scipy contains interesting classes for rotations which makes the calculations of rotation matrices easy.
3. Find the rotation error matrix which is the relative rotation between the given point1 and point2.
![Rerr](https://user-images.githubusercontent.com/79858941/134506024-55ad87b4-e252-44fd-9333-e7856efb8bed.png)
4. Map/convert the rotation error matrix into a scalar rotation error
![scalar](https://user-images.githubusercontent.com/79858941/134506063-a9c2f0c0-e943-41a5-9a37-1fc7da593ef8.png)

I have tested the gauge movement after that, and it looks better now. There is "_continuity_" in the gauge movement (the gauge moves for x, y, z movements and continues moving for rx and rz). 

@Bassim Would you please test enzel gauge, and see how it looks? If not so good, I need to tune the parameter multiplied by the trace to a value that is good for both meteor and enzel movements.